module-system/sov-modules-macros/build.rs

```diff
@@ -1,14 +1,59 @@
 use std::path::PathBuf;
 use std::{env, fs, io};
 
-fn main() -> io::Result<()> {
+use anyhow::{anyhow, Context};
+
+fn resolve_manifest_path() -> anyhow::Result<PathBuf> {
+    let manifest = "constants.json";
+    match env::var("CONSTANTS_MANIFEST"){
+        Ok(p) if p.is_empty() => {
+            Err(anyhow!(
+                "failed to read target path for sovereign manifest file: env var `CONSTANTS_MANIFEST` was set to the empty string"
+            ))
+        },
+        Ok(p) => PathBuf::from(&p).canonicalize().map_err(|e| {
+                anyhow!("failed to canonicalize path for sovereign manifest file from env var `{p}`: {e}")
+        }),
+        Err(_) => {
+            let mut current_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+            loop {
+                if current_path.join(manifest).exists() {
```

Comment:
Should we check if [is_file](https://doc.rust-lang.org/std/path/struct.Path.html#method.is_file) ? Also, we should probably canonicalize the `current_path`, so the `parent` backtrack will be unambiguous.

Reply:
Sure, both good ideas. Done!